### DISCUSSION TITLE
where is the retry for states?

### DISCUSSION THREAD

**Discussion by globalflea** [serverlessworkflow/specification#547] [Q&A]
I can find mentions of Retry for Actions, but i can't seem to find Retry for State - or how i specify the Retry policy for a state? Or are Retries only applicable to Actions?

**Comment by tsurdilo** [comment]
Currently only for actions. Can you provide an use case where a state-retry policy would be good to have?
Do you mean cases where you want to retry all actions inside an operation state for example based on the state data output? I think for those types of cases you can use switch state after operation state and transition back to operation state for certain condition.

**Reply by globalflea** [reply]
From the specs on StateExecTimeout :
"...single property. To show an example, let's say that we set the single property to "PT10S", meaning 10 seconds. A workflow state "X" which defines this timeout when first executed has the max execution timeout set to 10 seconds. *If the state execution is then retried, each time it is retried*, the individual retry max execution timeout is again 10 seconds." (My italics)

I may be understanding it wrong, but it seems to imply states can be retried?

**Comment by tsurdilo** [comment]
No you are not misunderstanding it, the text needs to be updated and clarified. Please open an issue and will work on it.
State retries can become fairly complex, for example if state timeout is specified, then actions inside the state are bound to it. If state timeout fires, what happens to actions that are still running? That in addition to not being able to currently define retryRef on state level, if indeed we want to keep state retries...Yeah, we need to figure this out - nice catch, thank you!
My advice is that if you are trying to implement this, to not implement state timeouts yet until we resolve this.
